### ISSUE TITLE
debug_panel can fail to track some queries depending on how ExistDB is imported

### ISSUE THREAD

**Issue by lddubeau** [emory-libraries/eulexistdb#8]
The code in `debug_panel.py` currently does the following in order to replace the stock `ExistDB` class with one that tracks queries:

    eulexistdb.db.ExistDB = ExistDBTracker

The problem is that this will affect only code that gets the new reference after this assignment. The code that has already grabbed a reference to `eulexistdb.db.ExistDB` won't be tracked. For instance, this will remain invisible to the debug panel:

```
[In the list of imports at the top of the module:]
from eulexistdb.db import ExistDB

[Later in the code:]
ExistDB(...)
```

The list of imports is resolved before the panel is enabled so it grabs a reference to the stock `ExistDB` and saves it locally as `ExistDB`. Later, when the panel is enabled, the local reference remains unaffected.

I don't see a solution to this short of modifying `ExistDB` itself to have provisions for tracking queries. One way to do it would be to dynamically patch *the class* so that all instances benefit from the patched code.

**Comment by rlskoeser** [comment]
I suspected there were probably cases where it wouldn't track the queries, but wasn't sure how to identify or resolve.

Since the debug panel and `ExistDB` are in the same codebase, maybe having tracking hooks in place that are turned off by default wouldn't be too bad.  And that could also address #7.

**Comment by rlskoeser** [comment]
Should be addressed by changes made for #7
